governance/src/main/java/org/apache/servicecomb/governance/handler/RetryHandler.java

```diff
@@ -65,21 +63,26 @@ protected Retry createProcessor(RetryPolicy policy) {
   private Retry getRetry(RetryPolicy retryPolicy) {
     LOGGER.info("applying new policy: {}", retryPolicy.toString());
 
-    List<Integer> statusList = Arrays.stream(retryPolicy.getRetryOnResponseStatus().split(","))
-        .map(Integer::parseInt).collect(Collectors.toList());
-
     RetryConfig config = RetryConfig.custom()
         .maxAttempts(retryPolicy.getMaxAttempts())
-        .retryOnResult(getPredicate(statusList))
+        .retryOnResult(getPredicate(retryPolicy.getRetryOnResponseStatus()))
         .retryExceptions(retryExtension.retryExceptions())
-        .waitDuration(Duration.ofMillis(retryPolicy.getWaitDuration()))
+        .intervalFunction(getIntervalFunction(retryPolicy))
         .build();
 
     RetryRegistry registry = RetryRegistry.of(config);
     return registry.retry(retryPolicy.getName());
   }
 
-  private Predicate<Object> getPredicate(List<Integer> statusList) {
+  private IntervalFunction getIntervalFunction(RetryPolicy retryPolicy) {
+    if (retryPolicy.getRetryStrategy().equals("RandomBackoff")) {
```

Comment:
Create a String constant in RetryPolicy, and use "RandomBackoff".equls(retryPolicy.getRetryStrategy())

Reply:
OK. i will modify this. But retryPolicy.getRetryStrategy() will not be nullpoint.